### DISCUSSION TITLE
Stuck in development for 4 days because of an invisible drop-down menu (check last post for context)

### DISCUSSION THREAD

**Discussion by FermiBrudja** [dialogic-godot/dialogic#850] [Q&A] [answered]
This is the first time I install Dialogic, also I'm new to Godot so I'm not sure if this is a bug or I made a mess of it because of my noobness.

I was following the Getting Started With Dialogic video on YouTube and followed the steps. I created a new theme to go with my own character and I changed its default font that comes with Dialogic to my font. Then I fiddled with the soundbits a little before noticing that my font looked too small and I wished to change its default size but I couldn't find how. 

When I looked back at the Default Theme, the theme had everything changed, with all the custom things I made on the new theme and I found it odd that it still said it used the Default Font when the preview was all the time showing my custom font added to the theme of my character. I pressed the button to create a new fresh theme and every time I do this it creates an exact copy of the new theme I made for my character. I can't modify these new themes I create (nothing updates even if I pick another font, other sounds, edit the dialog box or its text...) nor return the Default Theme to its default settings. I can only edit the theme of my character and this spreads into ALL the themes.

Is this a bug or did I do something wrong? This happened Friday and I can't seem to find a solution since then. The Settings use the Default Theme, I didn't change anything there at all 🤔

**Comment by FermiBrudja** [comment]
Trying to troubleshoot this by going out of context, I found out how to reproduce this "bug" and change the theme that goes by default no matter what you set in the Settings of Dialogic.

I went to my only character --the protag hero-- and under the Theme I changed it back to the Default Theme. By doing this, now ALL the themes changed to the Default Theme, including the one I made exclusively for my protag. If I swap it to the protag's theme, all the themes changes to my custom theme. And so on so forth...

I created the second protag just to see what would happen and made it to use the Default Theme, leaving my main protag to use his custom theme. Now all the themes use the second protag default theme and the one I use for the first protag is not used. 

What is more odd though is, if I press the Add Theme button, if you cycle through all the created themes (even the new untouched ones) it will randomly preview the themes for the first protag and the second protag at random.

In a nutshell, I can't possibly create a theme for each situation or character because every time I attempt to edit one, it will change to a random one from the previous characters and I need to spam the Preview Changes until it selects the matching theme I'm editing.

Is this intended? Is this a bug? Or I am using all of it wrong thus causing this loop of randomness to happen?

**Comment by FermiBrudja** [comment, marked as answer]
All right, nevermind that. Since I'm a noob and this piece of GUI isn't very "self-explanatory":
![image](https://user-images.githubusercontent.com/68450307/160407401-716dc79b-5d5f-44a2-b507-a8f4efdf9996.png)
Now I SEE I can click that invisible button and choose among a drop-down to preview themes based upon a list, instead of being a fixed preview by default.
Great... I made a dumb of myself. Rather, my UX got a seizure after realizing I could click that "invisible drop-down". There is no excuse now. I see that I must adjust the preview each time to match the new one, ON EVERY THEME. It doesn't even save it. This is going to be so fun to edit when I finish adding all the 85 characters with their respective themes! Will that drop-down break? I must try!

Going to change the title again to something more fitting, and I hope this bad UX serves to improve the addon. 4 days stuck because of an invisible drop-down that isn't able to save its selection for each individual theme is not very cool... 😭 Either eliminate it all together or make it save its selection for each theme and save further headaches.

Pah, close this topic, and sorry for any incovenience I might have caused with my noobsness... English isn't even my first language but I try to be respectful and polite and inform of this UX failure.

**Reply by Jowan-Spooner** [reply]
Thanks for the report. This should most certainly be changed.

**Reply by coppolaemilio** [reply]
I'll be working on the changes, this is the issue: https://github.com/coppolaemilio/dialogic/discussions/850
